src/gui/editorwidgets/qgscolorwidgetwrapper.cpp

```diff
@@ -75,5 +75,15 @@ bool QgsColorWidgetWrapper::valid() const
 void QgsColorWidgetWrapper::setValue( const QVariant& value )
 {
   if ( mColorButton )
-    mColorButton->setColor( QColor( value.toString() ) );
+  {
+    if ( value.isNull() )
```

Comment:
Actually, when the widget has a null color and the user click on the OK button, it's the string "#000000" which is really stored. Thus, I wanted to explicitely indicate to the user what color is going to be stored with the current context. But a the message bar on the top of the form says something like "Some fields are NULL (default values are used): field_color".